python/nav/web/ipam/api.py

```diff
@@ -35,17 +35,25 @@
 
 
 from rest_framework import serializers
-#from nav.models.fields import CIDRField
+# from nav.models.fields import CIDRField
 
 
 # Inspired by
 # http://blog.karolmajta.com/parsing-query-parameters-in-rest-framework/
 class SuggestParams(serializers.Serializer):
     prefix = serializers.CharField()
     n = serializers.IntegerField(default=10)
-    size = serializers.IntegerField(default=256)
+    prefixlen = serializers.IntegerField(default=24, min_value=1, max_value=128)
     offset = serializers.IntegerField(default=0)
 
+    def validate(self, data):
```

Comment:
Actually, this breaks normal operation of the suggest function. Whenver I enter a valid prefix length, the interface now crashes because `validate()` needs to return a validated data dictionary.

Reply:
fixed